sw/device/tests/alert_handler_lpg_sleep_mode_pings.c

```diff
@@ -0,0 +1,425 @@
+    flash_ctrl_testutils_counter_init_zero(&flash_ctrl, kCounterTestPhase);
+    test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+    rnd_num_iterations = flash_ctrl_testutils_counter_get(kCounterTestPhase);
+  }
+
+  // If this is the first iteration,
+  // set num_iterations to a random value between 8 and 32
+  if (rnd_num_iterations == 0) {
+    // 8 <= num_iters <= 32
+    rnd_num_iterations = rand_testutils_gen32_range(8, 32);
+    flash_ctrl_testutils_counter_set_at_least(&flash_ctrl, kCounterTestPhase,
+                                              rnd_num_iterations);
+    // LOG_INFO("phase1-num-iters = %d", rnd_num_iterations);
+  }
+
+  // Test phase #1: ping-timeout = 256
+  // No ping_timeout_fail alert shall be fired
+  // To ensure that the ping mechanism won't send spurious failure.
+  while (test_step_cnt < rnd_num_iterations) {
+    // Configure and lock the alerts
+    CHECK_DIF_OK(
+        dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+    if (!is_locked) {
+      alert_handler_config(/*ping_timeout=*/256);
+    }
+
+    // Power-on reset
+    if (pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) {
+      LOG_INFO("POR reset");
+      // Set  the AON timer to send a wakeup signal in ~150us.
+      // Go to deep sleep mode
+      aon_timer_testutils_wakeup_config(&aon_timer, 30);
+      enter_low_power(/*deep_sleep=*/(test_step_cnt % 2));
+    } else { /*wakeup reset*/
+      // Check the wakeup reason and do the cleanup
+      check_wakeup_reason();
+      cleanup_wakeup_src();
+
+      // After the wakeup, wait enough, and
+      // check that no ping-timeout-fail alert is fired
+      wait_enough_for_alert_ping();
+      CHECK_DIF_OK(dif_alert_handler_local_alert_is_cause(
+          &alert_handler, kDifAlertHandlerLocalAlertAlertPingFail, &is_cause));
+      CHECK(!is_cause,
+            "Expected response is ping_timeout_fail= 0! But we got "
+            "ping_timeout_fail = 1");
+
+      // Increment the test_step counter
+      flash_ctrl_testutils_counter_increment(&flash_ctrl, kCounterTestSteps);
+      test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+      // LOG_INFO("Test %d beginn", test_step_cnt);
+      // Set  the AON timer to send a wakeup signal in ~150us.
+      aon_timer_testutils_wakeup_config(&aon_timer, 30);
+      // Enter the normal sleep or deep sleep mode
+      enter_low_power(/*deep_sleep=*/(test_step_cnt % 2));
+      // enter_low_power(/*deep_sleep=*/rand_testutils_gen32_range(0,1));
+    }
+  }
+
+  // Test phase #2: ping_timeout = 2
+  // ping_timeout_fail alert shall be fired.
+  // To ensure that ping mechanism will continue to send out pings after wakeup
+  while (test_step_cnt < 2 * rnd_num_iterations) {
+    // Configure and lock the alerts
+    CHECK_DIF_OK(
+        dif_alert_handler_is_ping_timer_locked(&alert_handler, &is_locked));
+    if (!is_locked) {
+      alert_handler_config(/*ping_timeout=*/2);
+    }
```

Comment:
as mentioned above, it would be good to add a check here to make sure the alert handler is still locked after coming back from light sleep.